### PR TITLE
fix(filter): support filters on number fields

### LINKED ISSUE #1913: Number filters not working
Currently, filters for number properties are not working because the filter generator tries to create a label using the function `toLowerCase` which does not exist on numbers.

### PR DESCRIPTION
**Pull request by sleidig**
fixes #1913

### CODE REVIEW

**Review by tomwwinter — APPROVED** [review]
LGTM

**Review by sleidig — COMMENTED** [review]
revert config changes

File: src/app/core/config/config-fix.ts [hunk 1]
```diff
@@ -299,13 +299,15 @@ export const defaultJsonConfig = {
         "name",
         { id: "DisplayParticipantsCount", viewComponent: "DisplayParticipantsCount", label: $localize`Children` },
         "privateSchool",
-        "language"
+        "language",
+        "studentsNo"
```
Comment (sleidig) [inline src/app/core/config/config-fix.ts:303]:
```suggestion
        "language"
```

File: src/app/core/config/config-fix.ts [hunk 2]
```diff
@@ -299,13 +299,15 @@ export const defaultJsonConfig = {
...
       ],
       "filters": [
         {
           "id": "privateSchool",
           "label": $localize`Private School`
-        }
+        },
+        { "id": "studentsNo" }
```
Comment (sleidig) [inline src/app/core/config/config-fix.ts:310]:
```suggestion
        }
```

File: src/app/core/config/config-fix.ts [hunk 3]
```diff
@@ -322,7 +324,8 @@ export const defaultJsonConfig = {
               "component": "Form",
               "config": {
                 "fieldGroups": [
-                  { "fields": ["name", "privateSchool"] },
+                  { "fields": ["name", "privateSchool",
+                      "studentsNo"] },
```
Comment (sleidig) [inline src/app/core/config/config-fix.ts:328]:
```suggestion
                  { "fields": ["name", "privateSchool"] },
```

File: src/app/core/config/config-fix.ts [hunk 4]
```diff
@@ -808,6 +811,10 @@ export const defaultJsonConfig = {
         "dataType": "string",
         "label": $localize`:Label for the name of a school:Name`
       },
+      "studentsNo": {
+        "dataType": "number",
+        "label": "number"
+      },
```
Comment (sleidig) [inline src/app/core/config/config-fix.ts:817]:
```suggestion
```

**Pull request merged**